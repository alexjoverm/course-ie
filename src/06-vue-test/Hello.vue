
<template>
  <div>
    Hello {{ name }} from {{greeter}}!
    <button @click="sayHi">Say Hi</button>
  </div>
</template>

<script>
export default {
  props: ["greeter"],
  data() {
    return { name: "Vue" };
  },
  methods: {
    sayHi() {
      this.$emit("sayHi", "Hi!");
    },
  },
};
</script>
