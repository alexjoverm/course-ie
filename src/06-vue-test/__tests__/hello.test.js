import Vue from "vue";
import { mount, shallow } from "@vue/test-utils";
import Hello from "../Hello.vue";

/** Testing using plain Vue **/

// describe("Hello component", () => {
//   it("renders correctly", () => {
//     const comp = new Vue(Hello).$mount();
//     expect(comp.$el.textContent).toContain("Hello Vue");
//   });

//   it("renders correctly after changing the name state", () => {
//     const comp = new Vue(Hello).$mount();
//     comp.name = "Camel";

//     comp.$nextTick(() => {
//       expect(comp.$el.textContent).toContain("Hello Camel");
//     });
//   });

//   it("renders correctly passing the from prop", () => {
//     const propsData = {
//       greeter: "Joe",
//     };
//     const Constructor = Vue.extend(Hello);
//     const comp = new Constructor({
//       propsData,
//     }).$mount();

//     expect(comp.$el.textContent).toContain("Hello Vue from Joe");
//   });
// });

/** Testing rendered data using @vue/test-utils methods **/

// describe("Hello component", () => {
//   it("renders correctly", () => {
//     const comp = mount(Hello);
//     expect(comp.html()).toContain("Hello Vue");
//   });

//   it("renders correctly after changing the name state", () => {
//     const comp = mount(Hello);
//     comp.setData({ name: "Camel" });
//     expect(comp.html()).toContain("Hello Camel");
//   });

//   it("renders correctly passing the from prop", () => {
//     const propsData = { greeter: "Joe" };
//     const comp = mount(Hello, {
//       propsData,
//     });

//     expect(comp.html()).toContain("Hello Vue from Joe");
//   });
// })

/** Testing with Snapshot Testing **/

describe("Hello component", () => {
  it("renders correctly", () => {
    const comp = mount(Hello);
    expect(comp.html()).toMatchSnapshot();
  });

  it("renders correctly after changing the name state", () => {
    const comp = mount(Hello);
    comp.setData({ name: "Camelo" });
    expect(comp.html()).toMatchSnapshot();
  });

  it("renders correctly passing the from prop", () => {
    const propsData = { greeter: "Joe" };
    const comp = mount(Hello, { propsData });

    expect(comp.html()).toMatchSnapshot();
  });

  it("calls sayHi when click on button", () => {
    const stub = jest.fn();
    const methods = { sayHi: stub };
    const comp = mount(Hello, { methods });

    comp.find("button").trigger("click");
    expect(stub).toBeCalled();
  });

  it("sayHi emits a sayHi event", () => {
    const comp = mount(Hello);
    comp.vm.sayHi();
    expect(comp.emitted().sayHi).toBeTruthy();
  });
});
