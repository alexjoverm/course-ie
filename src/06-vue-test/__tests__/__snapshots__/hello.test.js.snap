// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello component renders correctly 1`] = `
<div>
  Hello Vue from !
  <button>Say Hi</button>
</div>
`;

exports[`Hello component renders correctly after changing the name state 1`] = `
<div>
  Hello Camel from !
  <button>Say Hi</button>
</div>
`;

exports[`Hello component renders correctly passing the from prop 1`] = `
<div>
  Hello Vue from Joe!
  <button>Say Hi</button>
</div>
`;
